File: 2020-10-30-client-libraries/node/authentication.js
// globally
// const stripe = require('stripe')('sk_test...');
//

// (async () => {
//   // Per request
//   const stripe = require('stripe')();
//   const customers = await stripe.customers.retrieve('cus_ICiIH7WIPI4Gr1', {
//     api_key: 'sk_test...'
//   });
//
//
//   console.log(customers);
// })()

// With connect
(async () => {
  const stripe = require('stripe')('sk_test...');
  const customer = await stripe.customers.retrieve('cus_HDfWzCQ6UEVtfu', {
    stripeAccount: 'acct_1Ey3h1BqeQ4DKpna'
  });

  console.log(customer);
})()

File: 2020-10-30-client-libraries/java/Authentication.java
import com.stripe.Stripe;
import com.stripe.param.*;
import com.stripe.model.*;
import com.stripe.net.*;

class Authentication {
    public static void main(String[] args) {
      try {
        // Globally
        // Stripe.apiKey = "sk_test...";

        // CustomerListParams params = CustomerListParams.builder().build();
        //
        // Per request
        // RequestOptions requestOptions = RequestOptions.builder()
        //   .setApiKey("sk_test...")
        //   .build();
        // Customer customer = Customer.retrieve("cus_ICiIH7WIPI4Gr1", requestOptions);

        // With connect and API key set per request
        //
        // RequestOptions requestOptions = RequestOptions.builder()
        //   .setStripeAccount("acct_1Ey3h1BqeQ4DKpna")
        //   .setApiKey("sk_test...")
        //   .build();
        // Customer customer = Customer.retrieve("cus_HDfWzCQ6UEVtfu", requestOptions);

        // With connect and API key set globally.
        Stripe.apiKey = "sk_test...";
        RequestOptions requestOptions = RequestOptions.builder()
          .setStripeAccount("acct_1Ey3h1BqeQ4DKpna")
          .build();
        // CustomerListParams params = CustomerListParams.builder().build();
        // CustomerCollection customers = Customer.list(params, requestOptions);
        Customer customer = Customer.retrieve("cus_HDfWzCQ6UEVtfu", requestOptions);

        System.out.println(customer);
      } catch(Exception e) {
        System.out.println(e);
      }
    }
}
